clients/src/test/java/org/apache/kafka/clients/consumer/KafkaConsumerTest.java

```diff
@@ -421,8 +479,152 @@ public void testWakeupWithFetchDataAvailable() {
+        responseMap.put(partition1, Errors.NONE.code());
+        client.prepareResponseFrom(new MockClient.RequestMatcher() {
+            @Override
+            public boolean matches(ClientRequest request) {
+                OffsetCommitRequest commitRequest = new OffsetCommitRequest(request.request().body());
+                OffsetCommitRequest.PartitionData partition1Data = commitRequest.offsetData().get(partition1);
+                // 5 records from topic1 partition have been returned to the consumer and auto commit is enabled;
+                // therefore, the committed offset of topic1 partition should be 5.
+                if (partition1Data.offset == 5) {
+                    commitAsExpected.set(true);
+                    return true;
+                }
+                return false;
+            }
+        }, offsetCommitResponse(responseMap), coordinator);
+
+        consumer.subscribe(Collections.singletonList(topic2));
+        assertTrue(commitAsExpected.get());
+    }
+
+    /**
+     * "consumer" subscribes to "topic1" (with only one partition) and has auto-commit disabled.
+     * "consumer" reads 5 records from the single partition of "topic1" from offset 0.
+     * "consumer" changes subscription to "topic2" (with only one partition).
+     * Since auto-commit is disabled, no offset commit request should be sent.
+     */
+    @Test
+    public void testSubscriptionChangeAndOffsetCommitWithAutoCommitDisabled() {
+        final String topic1 = "topic1";
+        final TopicPartition partition1 = new TopicPartition(topic1, 0);
+        final String topic2 = "topic2";
+
+        int sessionTimeoutMs = 30000;
+        int heartbeatIntervalMs = 3000;
+
+        // adjust auto commit interval lower than heartbeat so we don't need to deal with
+        // a concurrent heartbeat request
+        int autoCommitIntervalMs = 1000;
+
+        Time time = new MockTime();
+        MockClient client = new MockClient(time);
+        Map<String, Integer> tpCounts = new HashMap<>();
+        tpCounts.put(topic1, 1);
+        tpCounts.put(topic2, 1);
+        Cluster cluster = TestUtils.singletonCluster(tpCounts);
+        Node node = cluster.nodes().get(0);
+        client.setNode(node);
+        Metadata metadata = new Metadata(0, Long.MAX_VALUE);
+        metadata.update(cluster, time.milliseconds());
+        PartitionAssignor assignor = new RoundRobinAssignor();
+
+        final KafkaConsumer<String, String> consumer = newConsumer(time, client, metadata, assignor,
+                sessionTimeoutMs, heartbeatIntervalMs, false, autoCommitIntervalMs);
+        consumer.subscribe(Arrays.asList(topic1), getConsumerRebalanceListener(consumer));
+
+        prepareRebalance(client, node, assignor, partition1);
+        consumer.poll(0);
+
+        // respond to the outstanding fetch so that we have data available on the next poll
+        client.respondFrom(fetchResponse(partition1, 0, 5), node);
```

Comment:
nitpick: seems like the offsets don't actually matter in this test (and the similar one for unsubscribe(). Maybe we could leave fetching out of the test?